appunite/Loudius
# [LD-139] chore: test coverage report

# Pull-Request
## Description
- Add per pull request code coverage report generation. 

### Why?
 - There is no way to monitor the code coverage increase of decrease, in order to maintain high quality code, we want to have a tool allowing us to monitor the changes.
### What?
- Add a gradle configuration allowing to generate the report and add a task correctly configuring the coverage report settings.

## Links to related issues
(https://loudius.atlassian.net/browse/LD-139)

## Demo
<img width="973" alt="image" src="https://github.com/appunite/Loudius/assets/15973138/beb20f67-5104-416b-93fb-f1f92b33b9db">

## How to test
 - Try to add some code uncovered by unit tests
 - Push the changes, there should be a PR comment added indicating a decrease in code coverage

## Documentation
[Github action](https://github.com/marketplace/actions/jacoco-report) used to create a PR comment
[Jacoco](https://github.com/jacoco/jacoco) Library genereting the coverage report
## Checklist
- [x] Functionality is covered by unit tests ( not applicable )
- [x] Functionality is covered by integration tests ( not applicable )
